### DISCUSSION TITLE
Consider only supporting Tailwind

### DISCUSSION THREAD

**Discussion by dmyers** [rappasoft/laravel-livewire-tables#1574] [Q&A]
Have you considered only supporting Tailwind? Supporting Bootstrap seems to really bloat all of the templates files substantially and most people by now who use Livewire tend to use the TALL stack (T=Tailwind). So I'm very curious if anyone using this package even uses Bootstrap to warrant the time to maintain it.

I've personally removed all of the `isBootstrap()` code as it's really excessive when trying to customize the package, though I do recognize there is the ways to override the default classes in some cases it isn't enough.

**Comment by lrljoe** [comment]
There's a significant enough chunk of websites that use the package that use Bootstrap.

v4 will probably be Bootstrap-5 and Tailwind only.

It used (a while back) to be separate blades all over the place, but maintaining it all was just not realistic, thus the merging, and use of conditional classes.

There's a few upcoming changes for custom classes in more places.
There's also some upcoming changes to allow use of custom blades for certain elements, on a per-table basis.

Which bits are you not able to customise within the existing methods available that'd be useful?

I typically recommend against customising the views as much as possible, as things do change to break-fix, performance tweak, or introduce new features, although the number of changes per week has reduced a bit!

**Comment by giles247** [comment]
Please at least keep Bootstrap 5. Taiwind is horrible.

**Reply by dmyers** [reply]
By that logic, why are you using Livewire when jQuery exists? That's a pretty regressive viewpoint. Bootstrap had it's day and it's over for serious users.

**Reply by lrljoe** [reply]
> Please at least keep Bootstrap 5. Taiwind is horrible.

There's no plan to remove the support for bootstrap 5 or tailwind during v3, or during v4.  That's support for both for at least the next few years, if not well beyond that.

The plan is to simplify a lot of the DOM elements, move more of the default class determination into methods, so that there's no bloat from different options

**Reply by dmyers** [reply]
Great idea Joe! I love it.

**Comment by giles247** [comment]
> That's a pretty regressive viewpoint. Bootstrap had it's day and it's over for serious users.

For 'serious users'? That couldn't sound more arrogant if you tried. Sure Bootstrap may be old now, but Tailwind is not an improvement. Personally, I rarely use Bootstrap, but I would rather pick apart and re-style a supplied Bootstrap template than a Tailwind one. Anyway, choice is good, so thanks Joe for keeping Boostrap option for us non-elite and non-progressive developers 😁

**Reply by dmyers** [reply]
Sorry if that came off that way. That was poorly worded I apologize. What I was really trying to convey though is that there is a cost of time that Joe and other contributors have to invest on maintaining both. This package is already incredible, but that time could be spent on adding and improving it is all. It's really in my mind like supporting PHP 5. While I understand your view, the TALL stack is very powerful. I also used Bootstrap and even contributed to v1 in the early days. It's just part of the tech world, that things evolve and iterate, but again I worded it badly.

I just see this having a huge amount of potential and having to keep track of both Bootstrap's core classes and Tailwind's is a lot and will become technical debt for this project's growth. For example one of the reasons I had to customize many of the views was that I have my own button component and so I wanted the Columns dropdown and others to match mine that I used when I added custom buttons to the toolbar. Another is that is one area that when Livewire UI comes out, it would make sense to adopt that since this package is called Livewire Tables, would be Tailwind, so then Bootstrap would not make much sense.

**Comment by soap** [comment]
I use Bootstrap for some clients as they prefer.

**Comment by nerisonpitogo** [comment]
I am currently learning and I do not have any knowledge of tailwind. I am very weak in designing, so I rely only on ready-made bootstrap admin panels.
